lib/jetstream/api/consumer.ex

```diff
@@ -1,47 +1,230 @@
+    `:by_start_time`. It has the following options:
+       - `:instant` - the default policy. The messages will be pushed to the client as fast as possible.
+       - `:original` - the messages in the stream will be pushed to the client at the same rate that they
+         were originally received.
+
+  * `:sample_freq` - Sets the percentage of acknowledgements that should be sampled for observability, 0-100.
+    This value is a binary and for example allows both `30` and `30%` as valid values.
+  """
+
   import Jetstream.API.Util
 
-  @enforce_keys [:stream_name, :name]
-  defstruct stream_name: nil,
-            name: nil,
-            deliver_subject: nil,
-            deliver_policy: :all,
-            durable: true,
-            ack_policy: :explicit,
-            # 30sec
-            ack_wait: 30_000_000_000,
-            max_deliver: -1,
-            replay_policy: :instant,
-            opt_start_time: nil,
-            opt_start_seq: nil
+  @enforce_keys [:stream_name]
+  defstruct [
+    :backoff,
+    :deliver_group,
+    :deliver_subject,
+    :description,
+    :durable_name,
+    :filter_subject,
+    :flow_control,
+    :headers_only,
+    :idle_heartbeat,
+    :inactive_threshold,
+    :max_batch,
+    :max_expires,
+    :max_waiting,
+    :opt_start_seq,
+    :opt_start_time,
+    :rate_limit_bps,
+    :sample_freq,
+    :stream_name,
+    ack_policy: :explicit,
+    ack_wait: 30_000_000_000,
+    deliver_policy: :all,
+    max_ack_pending: 20_000,
+    max_deliver: -1,
+    replay_policy: :instant
+  ]
 
-  @type consumer_response :: %{
+  @type t :: %__MODULE__{
           stream_name: binary(),
-          name: binary(),
-          created: DateTime.t(),
-          config: consumer_config(),
-          delivered: %{
+          ack_policy: :none | :all | :explicit,
+          ack_wait: nil | non_neg_integer(),
+          backoff: nil | [non_neg_integer()],
+          deliver_group: nil | binary(),
+          deliver_policy:
+            :all | :last | :new | :by_start_sequence | :by_start_time | :last_per_subject,
+          deliver_subject: nil | binary(),
+          description: nil | binary(),
+          durable_name: nil | binary(),
+          filter_subject: nil | binary(),
+          flow_control: nil | boolean(),
+          headers_only: nil | boolean(),
+          idle_heartbeat: nil | non_neg_integer(),
+          inactive_threshold: nil | non_neg_integer(),
+          max_ack_pending: nil | integer(),
+          max_batch: nil | integer(),
+          max_deliver: nil | integer(),
+          max_expires: nil | non_neg_integer(),
+          max_waiting: nil | integer(),
+          opt_start_seq: nil | non_neg_integer(),
+          opt_start_time: nil | DateTime.t(),
+          rate_limit_bps: nil | non_neg_integer(),
+          replay_policy: :instant | :original,
+          sample_freq: nil | binary()
+        }
+
+  @type consumer_response :: %{
```

Comment:
I wonder if we really need this `consumer_` prefix for all types? Personally I would drop them.